src/server/zset_family.cc

```diff
@@ -130,6 +133,46 @@ OpResult<PrimeIterator> FindZEntry(const ZParams& zparams, const OpArgs& op_args
   return it;
 }
 
+// buf must be at least 12 bytes long.
+bool ToAsciiGeoHash(const std::optional<double>& val, char buf[]) {
```

Comment:
I'd take a `std::array<char, 12>*` and use it. It's safer, and doesn't use allocations. The only callsite only seems to be using an array, so use it there as well (what I'm trying to say is that it won't limit callers)